pkg/controller/podautoscaler/horizontal.go

```diff
@@ -182,45 +182,55 @@ func (a *HorizontalController) reconcileAutoscaler(hpa extensions.HorizontalPodA
 	cmStatus := ""
 	cmTimestamp := time.Time{}
 
-	if hpa.Spec.CPUUtilization != nil {
-		cpuDesiredReplicas, cpuCurrentUtilization, cpuTimestamp, err = a.computeReplicasForCPUUtilization(hpa, scale)
-		if err != nil {
-			a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeReplicas", err.Error())
-			return fmt.Errorf("failed to compute desired number of replicas based on CPU utilization for %s: %v", reference, err)
-		}
-	}
+	desiredReplicas := 0
+	timestamp := time.Now()
 
-	if cmAnnotation, cmAnnotationFound := hpa.Annotations[HpaCustomMetricsTargetAnnotationName]; cmAnnotationFound {
-		cmDesiredReplicas, cmStatus, cmTimestamp, err = a.computeReplicasForCustomMetrics(hpa, scale, cmAnnotation)
-		if err != nil {
-			a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeCMReplicas", err.Error())
-			return fmt.Errorf("failed to compute desired number of replicas based on Custom Metrics for %s: %v", reference, err)
+	if currentReplicas > hpa.Spec.MaxReplicas {
+		desiredReplicas = hpa.Spec.MaxReplicas
+	} else if hpa.Spec.MinReplicas != nil && currentReplicas < *hpa.Spec.MinReplicas {
+		desiredReplicas = *hpa.Spec.MinReplicas
+	} else if currentReplicas == 0 {
+		desiredReplicas = 1
+	} else {
```

Comment:
Because you can have a failure here. Plus if there is no replicas you will get 0 responses from heapster what means an error.